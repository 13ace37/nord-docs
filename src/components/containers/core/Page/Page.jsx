/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import styled from "styled-components";

import { motion } from "styles/theme";

import { baseBackgroundColor } from "../shared/styles";

/**
 * A basic wrapper component for page content.
 *
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/main
 * @since 0.3.0
 */
const Page = styled.main`
  background-color: ${baseBackgroundColor};
  transition: background-color ${motion.speed.duration.transition.base.themeModeSwitch}ms ease-in-out;
`;

export default Page;
