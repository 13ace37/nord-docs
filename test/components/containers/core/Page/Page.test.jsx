/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import React from "react";
import { render } from "react-testing-library";

import Page from "containers/core/Page";

test("snapshot", () => {
  const { container } = render(<Page />);
  expect(container).toMatchSnapshot();
});
