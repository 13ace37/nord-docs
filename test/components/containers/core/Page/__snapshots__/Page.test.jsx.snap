// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
.c0 {
  -webkit-transition: background-color 400ms ease-in-out;
  transition: background-color 400ms ease-in-out;
}

<main
  class="c0"
/>
`;
